src/main/webapp/app/programming-assessment/manual-result/programming-assessment-manual-result-dialog.component.ts

```diff
@@ -61,12 +75,22 @@ export class ProgrammingAssessmentManualResultDialogComponent implements OnInit
                 )
                 .subscribe(() => (this.isLoading = false));
         }
-        this.participation = this.result.participation! as StudentParticipation;
+        if (this.result.hasComplaint) {
+            this.getComplaint(this.result.id);
+            this.getParticipation();
```

Comment:
Can you explain to me, why we need this here? I cannot follow why
```ts
this.participation = this.result.participation! as StudentParticipation;
```
does not work in this case.

Reply:
What I discovered is that in case that we have complaint, we can't get from it's participation student's name. In order to get it I explicitly ask again for the participation

Reply:
Do we really want to get the students name? Or should it be hidden?

Reply:
I am not sure here. Maybe @krusche can give more insight. However, inside the manual result dialog we explicitly show student's name